### DISCUSSION TITLE
Buttons not functioning

### DISCUSSION THREAD

**Discussion by RubenStark** [heroui-inc/heroui#1139] [Feedback]
Im using next with react and js, and when I add a Button it doesnt function, i use StyledButton instead but it lacks some features

**Comment by jrgarciadev** [comment]
Hey @RubenStark, here's a codesanbox opened from the docs, make sure to have the same version and structure:

https://codesandbox.io/s/7plowf?file=%2FApp.js&from-sandpack=true

**Comment by RubenStark** [comment]
I am using it that way, the problem is the color property, everything works as it should but the color, I have to instert the className=”bg-blue-500” in order to have a bg color

Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: Junior ***@***.***>
Enviado: martes, 30 de mayo de 2023 03:37 p. m.
Para: ***@***.***>
CC: ***@***.***>; ***@***.***>
Asunto: Re: [nextui-org/nextui] Buttons not functioning (Discussion #1139)


Hey @RubenStark<https://github.com/RubenStark>, here's a codesanbox opened from the docs, make sure to have the same version and structure:

https://codesandbox.io/s/7plowf?file=%2FApp.js&from-sandpack=true

**Comment by Shayan-retrocket** [comment]
If you're using Next.js with React and JavaScript, and you're facing an issue where a regular `<Button>` component doesn't work as expected, you can try a few troubleshooting steps to identify and resolve the problem. Here are a few suggestions:

1. Check your import statement: Make sure you're importing the correct `<Button>` component from the library you're using. For example, if you're using Material-UI, the import statement should look like this: `import { Button } from '@mui/material';`

2. Verify event handlers: Ensure that you have added event handlers, such as `onClick`, to your `<Button>` component and that they are functioning correctly. Double-check the event binding syntax and any associated function logic.

3. Review component placement: Ensure that you have placed the `<Button>` component in the appropriate location within your component hierarchy. Verify that it is properly nested within the component tree and has access to the necessary props.

4. Confirm state management: If your `<Button>` component relies on state changes, make sure you're managing the state correctly. Check that state variables are defined, updated, and passed to the component as props when necessary.

If after trying these troubleshooting steps the issue persists, you may need to provide more specific information or code examples to help identify the problem. Regarding the alternative `StyledButton` component you mentioned, it's important to note that its lack of certain features may be due to its specific implementation or customization. You might need to customize it further or explore other button component libraries that offer the desired functionality.
